pareto/utilities/earthquake_distance.py

```diff
@@ -0,0 +1,230 @@
+    return date_ts
+
+
+def calculate_earthquake_distances(
+    swd_latlons,
+    api="usgs",
+    max_radius_mi=5.59,
+    min_magnitude=3,
+    min_date=None,
+    max_date=None,
+    save=None,
+    overwrite=False,
+):
+    # swd_latlons is a list of dicts with id, lat, and lon
+    if api not in ("usgs", "texnet"):
+        raise "api must be either usgs or texnet"
+
+    min_date_ts = convert_date_to_timestamp(min_date)
+    max_date_ts = convert_date_to_timestamp(max_date)
+    if max_date_ts:
+        max_date_ts += 24 * 60 * 60  # max date 24:00:00
+
+    if min_date_ts and max_date_ts and min_date_ts > max_date_ts:
+        raise "min_date must be earlier than or equal to max_date"
+
+    if save:
+        m = save_re.match(save)
+        if not m:
+            raise "save must be in *.csv or *.xlsx"
+        fmt = m[1].lower()
+        if not overwrite and os.path.exists(save):
+            raise f"{save} already exists"
+    else:
+        fmt = None
+
+    earthquake_distances = []
+
+    keys = ("swd_id", "eq_id", "time", "distance_mi", "magnitude")
+
+    for swd_latlon in swd_latlons:
+        swd_id = swd_latlon["id"]
+        swd_lat = swd_latlon["lat"]
+        swd_lon = swd_latlon["lon"]
+        if api == "usgs":
+            max_radius_km = max_radius_mi / mi_per_m / 1000  # 5.59 mi
+            url = usgs_api_url.format(
+                lat=swd_lat,
+                lon=swd_lon,
+                max_radius_km=max_radius_km,
+                min_magnitude=min_magnitude,
+            )
+        else:
+            url = texnet_api_url.format(
+                lat=swd_lat,
+                lon=swd_lon,
+                max_radius_mi=max_radius_mi,
+                min_magnitude=min_magnitude,
+            )
+
+        try:
+            request = urllib.request.Request(url)
+            with urllib.request.urlopen(request) as f:
+                response = json.load(f)
+        except:
+            raise
+        for feat in response["features"]:
+            eq_id = feat["id"]
+            props = feat["properties"]
+            coords = feat["geometry"]["coordinates"]
+
+            time_ts = (props["time"] if api == "usgs" else props["Event_Date"]) / 1000
+            if (min_date_ts and time_ts < min_date_ts) or (
+                max_date_ts and time_ts > max_date_ts
+            ):
+                continue
+
+            time = datetime.fromtimestamp(time_ts).strftime("%Y-%m-%d %H:%M:%S")
+            lat = coords[1]
+            lon = coords[0]
+            mag = props["mag"] if api == "usgs" else props["Magnitude"]
+            dist_mi = geod.line_length([swd_lon, lon], [swd_lat, lat]) * mi_per_m
+            earthquake_distances.append(
+                {
+                    keys[0]: swd_id,
+                    keys[1]: eq_id,
+                    keys[2]: time,
+                    keys[3]: dist_mi,
+                    keys[4]: mag,
+                }
+            )
+
+    if fmt == "csv":
```

Comment:
I'm finding that the csv files generated have every other line blank, e.g.:

![image](https://github.com/project-pareto/project-pareto/assets/26798258/dec1c924-26c6-4c44-8b49-d4592459b439)

Is there a way to modify this so that we don't have the blank lines?

Reply:
@tarnold17 Where do you see this problem? I'm using Windows newlines and it works for me.
![image](https://github.com/project-pareto/project-pareto/assets/7456117/0b124339-1c4a-4e29-8040-c5a11cff02cd)

Reply:
The above screenshot is from VIM (^M is a Windows newline). This is from Notepad in Windows:
![image](https://github.com/project-pareto/project-pareto/assets/7456117/10e9d135-4131-4dd6-a49f-b10a23b83446)

Reply:
@HuidaeCho my results are from running `python earthquake_distance.py` and then opening the file that gets created `eq_dist_texnet_results.csv`. The screenshot I posted is from Notepad++. Here's what it looks like for me opening in Excel and Notepad:

![image](https://github.com/project-pareto/project-pareto/assets/26798258/dd5aa8ad-034c-4e70-af63-d08fe43da261)

![image](https://github.com/project-pareto/project-pareto/assets/26798258/ab84f36f-f09f-4007-9220-d53c3af97ef6)

So the issue seems apparent when viewing in Notepad++ or Excel, but not in Notepad. I am not entirely sure what to make of that.

Reply:
@HuidaeCho here's what I see in Notepad++ when I turn on the "Show all characters" option: 
![image](https://github.com/project-pareto/project-pareto/assets/26798258/b0716f40-62bd-4318-a2d5-8e1845033d3d)

Reply:
Got it. CR,CR,LF needs to be fixed to CR,LF.